server/src/test/java/org/cloudfoundry/identity/uaa/authentication/manager/PasswordGrantAuthenticationManagerTest.java

```diff
@@ -354,7 +360,47 @@ void testOIDCPasswordGrantProviderNoRelyingPartyCredentials() {
             instance.authenticate(auth);
             fail();
         } catch (ProviderConfigurationException e) {
-            assertEquals("External OpenID Connect provider configuration is missing relyingPartyId or relyingPartySecret.", e.getMessage());
+            assertEquals("External OpenID Connect provider configuration is missing relyingPartyId.", e.getMessage());
+        }
+    }
+
+    @Test
+    void testOIDCPasswordGrantProviderJwtClientCredentials() throws MalformedURLException {
```

Comment:
Any way to simplify the rather large setup for this test?

Reply:
refactored this. other tests could also now re-use the mock methods, but in all other tests similar setup is done always from scratch